sound/soc/sof/ipc4-pcm.c

```diff
@@ -191,6 +191,70 @@ sof_ipc4_update_pipeline_state(struct snd_sof_dev *sdev, int state, int cmd,
  * prepare ioctl before the START trigger.
  */
 
+static int sof_ipc4_chain_dma_send_msg(struct snd_sof_dev *sdev,
+				       struct snd_sof_pcm_stream_pipeline_list *pipeline_list,
+				       bool allocate, bool enable)
+{
+	struct sof_ipc4_msg msg = {{ 0 }};
+	int i;
+
+	msg.primary = SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_GLB_CHAIN_DMA);
+	msg.primary |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
+	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_FW_GEN_MSG);
+
+	/*
+	 * Copy the Link DMA ID, host DMA ID, fifo_size and the SCS bit settings from the pipelines.
+	 * There are two pipelines involved, where the data is collected. The link side pipeline and
+	 * the host side pipeline. Part of the data needed is collected from the link side to the
```

Comment:
The link side pipeline and the host side pipeline.

Missing verb/complement or this needs to be part of the previous sentence. 

"part of the data" used in two different contexts sounds odd

Consider rewording, e.g.

To set-up the DMA chain, the host DMA ID and SCS setting are retrieved from the host pipeline configuration. Likewise the  link DMA ID and fifo_size are retrieved from the link pipeline configuration.

Reply:
Must have been really tired when writing this piece of comment. I take you version, thanks.